api/src/test/java/org/openmrs/util/ConceptReferenceRangeUtilityTest.java

```diff
@@ -0,0 +1,248 @@
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfAgeIsNoMatch() {
+		calendar.add(Calendar.YEAR, -2);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "#set( $criteria = $patient.getAge() >= 15 && $patient.getAge() <= 50 )$criteria";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfCriteriaIsInvalid() {
+		calendar.add(Calendar.YEAR, -1);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "invalidCriteria";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnTrueIfCriteriaIsEmpty() {
+		calendar.set(2019, Calendar.JUNE, 2);
+		person.setBirthdate(calendar.getTime());
+		String criteria = "";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfPersonIsNull() {
+		String criteria = "#set( $criteria = $patient.getAge() >= 15 && $patient.getAge() <= 50 )$criteria";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeInRange_shouldReturnFalseIfAgeOfAPersonIsNull() {
+		String criteria = "#set( $criteria = $patient.getAge() >= 15 && $patient.getAge() <= 50 )$criteria";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testGenderMatch_shouldReturnTrueIfGenderMatches() {
+		person.setGender("M");
+		String criteria = "#set( $criteria = $patient.getGender().equals('M') )$criteria";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testCriteriaWithPerson_shouldReturnTrueIfGenderMatches() {
+		person.setGender("M");
+		String criteria = "#set( $criteria = $patient.getGender().equals('M') )$criteria";
+
+		assertTrue(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testGenderMatch_shouldReturnFalseIfGenderDoesNotMatch() {
+		person.setGender("F");
+		String criteria = "#set( $criteria = $patient.getGender().equals('M') )$criteria";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testGenderMatch_shouldReturnFalseIfGenderIsNull() {
+		String criteria = "#set( $criteria = $patient.getGender().equals('M') )$criteria";
+
+		assertFalse(ConceptReferenceRangeUtility.evaluateCriteria(criteria, person));
+	}
+
+	@Test
+	public void testAgeAndGenderMatch_shouldReturnTrueIfAgeAndGenderMatch() {
+		calendar.add(Calendar.YEAR, -5);
+		person.setBirthdate(calendar.getTime());
+		person.setGender("M");
+		String criteria = "#set( $criteria = $patient.getAge() > 1 && $patient.getAge() < 10 && $patient.getGender().equals('M') )$criteria";
```

Comment:
Let us clean up all this criteria by removing the repeated `#set( $criteria` and `)$criteria` at the end.

Reply:
The criteria fails if I remove them.

Reply:
Yes i expect it to fail until you move them to the criteria evaluation utility class.

Reply:
I see what you mean